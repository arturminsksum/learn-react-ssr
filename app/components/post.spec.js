import React from 'react';
import Post from './post';

const mockPost = {
  id: 'abc-news',
  source: 'ABC News',
  author: 'ABC News',
  title: 'WATCH: What to know about the water challenge',
  description:
    "Dr. Jennifer Ashton kicks off a month-long 'Water Challenge' to look at how drinking more water can affect your health.",
  url: 'http://abcnews.go.com/GMA/video/water-challenge-52783678',
};

describe('Post', () => {
  const wrapper = mount(<Post item={mockPost} />);

  it('should render Post correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render post title', () => {
    expect(wrapper.find('.title').text()).toEqual(
      'WATCH: What to know about the water challenge',
    );
  });
});
