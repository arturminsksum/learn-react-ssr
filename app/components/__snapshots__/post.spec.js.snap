// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post should render Post correctly 1`] = `
<Component
  item={
    Object {
      "author": "ABC News",
      "description": "Dr. Jennifer Ashton kicks off a month-long 'Water Challenge' to look at how drinking more water can affect your health.",
      "id": "abc-news",
      "source": "ABC News",
      "title": "WATCH: What to know about the water challenge",
      "url": "http://abcnews.go.com/GMA/video/water-challenge-52783678",
    }
  }
>
  <article
    className="media box"
  >
    <div
      className="media-content"
    >
      <div
        className="content"
      >
        <a
          className="title"
          href="http://abcnews.go.com/GMA/video/water-challenge-52783678"
        >
          WATCH: What to know about the water challenge
        </a>
        <p>
          <span>
            Dr. Jennifer Ashton kicks off a month-long 'Water Challenge' to look at how drinking more water can affect your health.
          </span>
        </p>
        <p>
          <strong>
            Source: 
            ABC News
             
          </strong>
          <small>
            Author: 
            ABC News
             
          </small>
          <small>
            Time: 
          </small>
        </p>
      </div>
    </div>
  </article>
</Component>
`;
